### PR TITLE
chore(travis): Run validate on travis and report coverage

### LINKED ISSUE #27: Setup Travis CI
We need Travis to `npm run validate`. 

Blockers:
- [ ] #35  for node compat

### PR DESCRIPTION
**Pull request by kentcdodds**
Closes #27 

Thoughts @jonathanewerner and @sarbbottam?

My guess is that this will fail for node `0.10`, but there are still a TON of people using that version and I'd love this to work for them too. I don't expect it to be too much work because we're transpiling anyway :-) But we may need to change a few things in the code to make it work.

I think we can merge this and work on those issues individually.

### CODE REVIEW

**Comment by kentcdodds** [comment]
Looks like our first task after this gets merged is to update anywhere that we say: `import foo from '.'` to say: `import foo from './index'` because Node 0.10 doesn't support the dot reference.

Or maybe there's a babel plugin we could use (which would be sick). I wonder if there's a babel preset for Node 0.10. That would be a amazing...

**Comment by jonathanewerner** [comment]
Yey cool! I don't have pain with replacing those dots with `index`. 
Should that refactoring take place on this branch or should I merge this?

**Comment by sarbbottam** [comment]
>> Looks like our first task **after this gets merged** is to update anywhere that we say: import foo from '.' to say: import foo from './index' because Node 0.10 doesn't support the dot reference.

Should we have a PR that replaces `import foo from '.' ` to `import foo from './index'` merged first before this PR? 
I would ideally prefer to only merge a PR that would pass the build. :smile:

**Comment by jonathanewerner** [comment]
I'm with you there. :)

**Comment by kentcdodds** [comment]
> I would ideally prefer to only merge a PR that would pass the build.

`¯\_(ツ)_/¯` I'm good with that :-)

File: package.json [hunk 1]
```diff
@@ -40,7 +41,7 @@
   "author": "Kent C. Dodds <kent@doddsfamily.us> (http://kentcdodds.com/)",
   "license": "MIT",
   "dependencies": {
-    "configuration-validator": "1.0.0-beta.8",
+    "configuration-validator": "1.0.0-beta.9",
```
Comment (kentcdodds) [inline package.json]: This should fix some issues with Node 0.10

**Comment by kentcdodds** [comment]
[Looks like](https://travis-ci.org/kentcdodds/webpack-validator/jobs/116965612) one of our validators is using a newer `path` API. We'll need a workaround.

**Comment by sarbbottam** [comment]
Could we use https://www.npmjs.com/package/path-parse?

```js
import pathParse from 'path-parse`

if (!path.parse) {
  path.parse = pathParse
}
```

**Comment by kentcdodds** [comment]
I would prefer monkey-patching and just use `pathParse` rather than `path.parse` personally.

**Comment by jonathanewerner** [comment]
Opened #35 for this pending sub-task, would be a good first-timer I think. @kentcdodds not sure I'm correctly parsing your preference here: would like to prefer to use pathParse directly or conditionally overwrite ( = monkey-patch ? :)) path.parse with said function?

**Comment by kentcdodds** [comment]
I would prefer that we use it directly. I don't want to monkey patch it. :-)

**Comment by kentcdodds** [comment]
Rebased with the latest from master. Let's see if travis passes this time :-)

**Comment by kentcdodds** [comment]
ok, I think I've got it now :D

File: tests/passing-configs/formly.development.js [hunk 2]
```diff
@@ -91,22 +90,3 @@ function getCommonPlugins() {
     })
   ]);
 }
-
-function getBanner(env) {
```
Comment (kentcdodds) [inline tests/passing-configs/formly.development.js]: we really don't need this :)

File: package.json [hunk 3]
```diff
@@ -45,9 +47,9 @@
     "babel-preset-es2015": "6.5.0",
     "babel-preset-stage-2": "6.5.0",
     "babel-register": "6.5.2",
+    "codecov": "1.0.1",
     "compression-webpack-plugin": "0.3.0",
     "copy-webpack-plugin": "1.1.1",
-    "deindent": "0.1.0",
```
Comment (kentcdodds) [inline package.json]: This doesn't support Node 0.10 and we don't actually need it for anything.

**Comment by kentcdodds** [comment]
QCbsHcryxPQGc

**Pull request merged**